src/services/geo-location/geo.location.service.ts

```diff
@@ -0,0 +1,115 @@
+import { CACHE_MANAGER, Inject, Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { AxiosError } from 'axios';
+import { Cache } from 'cache-manager';
+import { GeoLocationCacheMetadata } from '@services/geo-location/geo.location.cache.metadata';
+import {
+  GeoServiceErrorException,
+  GeoServiceLimitReachedException,
+  GeoServiceNotAvailableException,
+} from '@common/exceptions/geo';
+import { LogContext } from '@common/enums';
+import { GeoInformation } from './geo.information';
+import { GeoPluginResponse } from './geo.plugin.response';
+
+const endpoint = 'http://www.geoplugin.net/json.gp?ip=';
```

Comment:
Move these entries to config, it is not impossible for them to change the endpoint or rate limits. We shouldn't need to redeploy then...